File: test/Ch13_StatisticalDescriptionOfData/student_ttest/different_means/test_ttest.jl
#=
    Unit tests of the ttest function

    Expected values obtained from HypothesisTests package
=#
using HypothesisTests


@testset "test_ttest_means_0" begin
    #=
        Checks against HypothesisTests
    =#
    data1 = [1, 22, 3, 4, 5, 16, 7, 8, 9]
    data2 = [1.1, 2, 3, 4, 5, 6, 7, 18, 9]

    # Student's "t"
    calculated_value = ttest(data1, data2)[1]
    expected_value = EqualVarianceTTest(data1, data2).t

    @test isapprox(calculated_value, expected_value; atol=1e-6)

    # Student's "significance"
    calculated_value = ttest(data1, data2)[2]
    expected_value = pvalue(EqualVarianceTTest(data1, data2))

    @test isapprox(calculated_value, expected_value; atol=1e-6)
end


@testset "test_ttest_means_1" begin
    #=
        Checks against HypothesisTests
    =#
    data1 = [1, 22, 3, 4, 5, 16]
    data2 = [1.1, 2, 3, 4, 5, 6, 7, 18, 9]
    # Student's "t"
    calculated_value = ttest(data1, data2)[1]
    expected_value = EqualVarianceTTest(data1, data2).t

    @test isapprox(calculated_value, expected_value; atol=1e-6)

    # Student's "significance"
    calculated_value = ttest(data1, data2)[2]
    expected_value = pvalue(EqualVarianceTTest(data1, data2))

    @test isapprox(calculated_value, expected_value; atol=1e-6)
end

File: test/Ch13_StatisticalDescriptionOfData/student_ttest/different_means/test_ttest.py
"""
    Unit tests of the Student t-test for difference of means

    Expected values obtained from SciPy stats library
"""
import pytest
import numericalrecipes as nr
import numpy as np
from scipy.stats import ttest_ind
from numpy.testing import assert_almost_equal


def test_ttest_0():
    """
        Checks against SciPy function
    """
    data1 = np.array([1, 22, 3, 4, 5, 16, 7, 8, 9])
    data2 = np.array([1.1, 2, 3, 4, 5, 6, 7, 18, 9])

    # Student's "t"
    calculated_value = nr.ttest(data1, data2)[0]
    expected_value = ttest_ind(data1, data2)[0]

    assert_almost_equal(calculated_value, expected_value, 4)

    # Student's "significance"
    calculated_value = nr.ttest(data1, data2)[1]
    expected_value = ttest_ind(data1, data2)[1]

    assert_almost_equal(calculated_value, expected_value, 4)


def test_ttest_1():
    """
        Checks against SciPy function
    """
    data1 = np.array([1, 22, 3, 4, 5, 16])
    data2 = np.array([1.1, 2, 3, 4, 5, 6, 7, 18, 9])

    # Student's "t"
    calculated_value = nr.ttest(data1, data2)[0]
    expected_value = ttest_ind(data1, data2)[0]

    assert_almost_equal(calculated_value, expected_value, 4)

    # Student's "significance"
    calculated_value = nr.ttest(data1, data2)[1]
    expected_value = ttest_ind(data1, data2)[1]

    assert_almost_equal(calculated_value, expected_value, 4)
